### PR TITLE
Added strategies to support a dinamic docker.sock location

### LINKED ISSUE #4: Support more docker.sock locations
Currently, `/var/run/docker.sock` location is resolved.

Let's add support for:

- [ ] `$XDG_RUNTIME_DIR/docker.sock` or `$HOME/.docker/run/docker.sock` - Linux (Rootless)
- [ ] `$HOME/.docker/desktop/docker.sock` - Linux (Docker Desktop)
- [ ] `$HOME/.docker/run/docker.sock` - Mac (Rootless)

### PR DESCRIPTION
**Pull request by cristianpalomino**
Given following requirements https://github.com/cristianpalomino/testcontainers-swift/issues/4

This PR wants to cover most of them

/var/run/docker.sock location ✅
 $XDG_RUNTIME_DIR/docker.sock ❌  or $HOME/.docker/run/docker.sock - Linux (Rootless) ✅
 $HOME/.docker/desktop/docker.sock - Linux (Docker Desktop)  ❌
 $HOME/.docker/run/docker.sock - Mac (Rootless) ✅

### CODE REVIEW

**Comment by eddumelendez** [comment]
Let's add a workflow similar to https://github.com/testcontainers/testcontainers-java/blob/main/.github/workflows/ci-rootless.yml#L43-L54 to test at least the rootles in linux

File: Sources/Testcontainers/Strategies/Docker+ClientStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+//
+
+import Foundation
+import Logging
+import NIO
+
+final class DockerClientStrategy {
+    
+    var strategies: [DockerClientStrategyProtocol] = [
+        TestcontainersStrategy(),
```
Comment (eddumelendez) [inline Sources/Testcontainers/Strategies/Docker+ClientStrategy.swift:15]: looks like it is not working. Desktop app is initialized and the connection is already stablished but tests can not run against it

Comment (cristianpalomino) [reply Sources/Testcontainers/Strategies/Docker+ClientStrategy.swift:15]: Pull and try again

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
🚀

**Pull request merged**